### PR TITLE
[mildom] update extractor to comply with current proxy

### PR DESCRIPTION
**Pull request by nao20010128nao**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

note that I copied the code from [here](https://github.com/nao20010128nao/ytdl-patched/blob/master/youtube_dl/extractor/mildom.py), which is my own work.

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This PR fixes the extractor to comply with current proxy.

### CODE REVIEW

File: yt_dlp/extractor/mildom.py [hunk 1]
```diff
@@ -172,9 +173,8 @@ class MildomVodIE(MildomBaseIE):
     _VALID_URL = r'https?://(?:(?:www|m)\.)mildom\.com/playback/(?P<user_id>\d+)/(?P<id>(?P=user_id)-[a-zA-Z0-9]+)'
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
-        m = self._VALID_URL_RE.match(url)
-        user_id = m.group('user_id')
+        m = self._valid_url_re().match(url)
```
Comment (pukkandan) [inline yt_dlp/extractor/mildom.py:176]:
```suggestion
        m = re.match(self._VALID_URL, url)
```

**Review by pukkandan — COMMENTED** [review]
_valid_url_re() is not a valid function

File: yt_dlp/extractor/mildom.py (hunk 1 above)
Comment (nao20010128nao) [reply yt_dlp/extractor/mildom.py:176]: sorry, I forgot to replace this

File: yt_dlp/extractor/mildom.py [hunk 2]
```diff
@@ -139,24 +140,24 @@ def _real_extract(self, url):
...
         m3u8_url = update_url_query(servers['stream_server'] + '/%s_master.m3u8' % video_id, stream_query)
-        formats = self._extract_m3u8_formats(m3u8_url, video_id, 'mp4', headers={
+        formats = self._extract_m3u8_formats(m3u8_url, result_video_id, 'mp4', headers={
             'Referer': 'https://www.mildom.com/',
             'Origin': 'https://www.mildom.com',
         }, note='Downloading m3u8 information')
         del stream_query['streamReqId'], stream_query['timestamp']
         for fmt in formats:
-            # Uses https://github.com/nao20010128nao/bookish-octo-barnacle by @nao20010128nao as a proxy
+            # source code behind bookish-octo-barnacle.vercel.app is here: https://github.com/nao20010128nao/bookish-octo-barnacle/
```
Comment (pukkandan) [inline yt_dlp/extractor/mildom.py:150]:
```suggestion
            # Uses https://github.com/nao20010128nao/bookish-octo-barnacle by @nao20010128nao as a proxy
```

**Pull request merged**